server-spi/src/main/java/org/keycloak/credential/CredentialTypeMetadata.java

```diff
@@ -0,0 +1,282 @@
+     * This label will reference this particular authenticator type.
+     * It should be clear to end users. For example, implementations can return "Authenticator Application" for OTP or "Security Key" for WebAuthn.
+     *
+     * Alternatively, this method can return a message key, so that it is possible to localize it for various languages.
+     */
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    /**
+     * @return the text, which will be shown to the user on various screens, like login screen with available authentication mechanisms.
+     * This text will reference this particular authenticator type.
+     * For example for OTP, the returned text could be "Enter a verification code from authenticator application" .
+     *
+     * Alternatively, this method can return a message key, so that it is possible to localize it for various languages.
+     */
+    public String getHelpText() {
+        return helpText;
+    }
+
+    /**
+     * Return the icon CSS, which can be used to display icon, which represents this particular authenticator.
+     *
+     * The icon will be displayed on various places. For example the "Select authenticator" screen during login, where user can select from
+     * various authentication mechanisms for two-factor or passwordless authentication.
+     *
+     * The returned value can be either:
+     * - Key of the property, which will reference the actual CSS in the themes.properties file. For example if you return "kcAuthenticatorWebAuthnClass"
+     *   from this method, then your themes.properties should have the property like for example "kcAuthenticatorWebAuthnClass=fa fa-key list-view-pf-icon-lg" .
+     *   This would mean that "fa fa-key list-view-pf-icon-lg" will be the actual CSS used.
+     * - the icon CSS class directly. For example you can return "fa fa-key list-view-pf-icon-lg" directly for the above example with WebAuthn.
+     *   This alternative is fine just if your authenticator can use same CSS class for all the themes.
+     *
+     * If you don't expect your authenticator to need icon (for example it will never be shown in the "select authenticator" screen), then
+     * it is fine to keep the default value.
+     */
+    public String getIconCssClass() {
+        return iconCssClass;
+    }
+
+    /**
+     * @return the providerID of the required action, which can be used by the user to create new credential of our type. Null if there is no
+     * action for creating credential. For example we're creating credential in case of "otp" type, but we're updating credential
+     * in case of type "password"
+     */
+    public String getCreateAction() {
+        return createAction;
+    }
+
+    /**
+     * @return the providerID of the required action, which can be used by the user to update credential of our type. Null if there is no
```

Comment:
Same here.  Not really a providerID.  It's the action used to trigger the AIA.  In the case of password, it's "UPDATE_PASSWORD".

Reply:
Yes, the "UPDATE_PASSWORD" is the providerId of requiredAction for update password - it's java class is UpdatePassword. Same case like for "createAction".

However if you have any better idea to this javadoc, we can discuss it.

Reply:
NP